Mathlib/Algebra/Associated.lean

```diff
@@ -420,6 +420,11 @@ protected def setoid (α : Type*) [Monoid α] :
   iseqv := ⟨Associated.refl, Associated.symm, Associated.trans⟩
 #align associated.setoid Associated.setoid
 
+theorem map {M N : Type*} [Monoid M] [Monoid N] (f : M →* N) {x y : M}
```

Comment:
Can you write it using `MonoidHomClass`? Thanks!

Reply:
Done. Please check that if the function prototype is correct.

Reply:
LGTM, let's just wait for the build to finish.